### PR TITLE
New eslint rules auto fix

### PR DESCRIPTION
**Pull request by akhenry**
Partially addresses https://github.com/nasa/openmct/issues/3056 by enabling new ESLint rules that require code changes that can be automatically applied using `npm run lint:fix`.

This is a huge code delta, so I'd recommend reading the associated documentation on the new rules, and skimming some of the code to see how it manifests in our code. I tried pushing smaller commits containing subsets of the rules, but some of the rules interact with each other in ways that mean that enabling them separately results in different code to if they were enabled at the same time.

These are the ESLint rules that are enabled here:
* https://eslint.org/docs/rules/no-implicit-coercion
* https://eslint.org/docs/rules/no-multi-spaces
* https://eslint.org/docs/rules/semi
* https://eslint.org/docs/rules/key-spacing
* https://eslint.org/docs/rules/keyword-spacing
* https://eslint.org/docs/rules/comma-spacing
* https://eslint.org/docs/rules/no-multiple-empty-lines
* https://eslint.org/docs/rules/no-unneeded-ternary
* https://eslint.org/docs/rules/no-whitespace-before-property
* https://eslint.org/docs/rules/object-curly-newline
* https://eslint.org/docs/rules/object-property-newline
* https://eslint.org/docs/rules/brace-style
* https://eslint.org/docs/rules/operator-linebreak
* https://eslint.org/docs/rules/padding-line-between-statements
* https://eslint.org/docs/rules/space-infix-ops
* https://eslint.org/docs/rules/space-unary-ops
* https://eslint.org/docs/rules/arrow-spacing

### Dependent Pull Requests
https://github.com/nasa/openmct/pull/3057 should be merged before this.

### Testing notes
This change touches many files. The changes are to code formatting, and should not have any functional impact. That said, testers should verify that the application loads properly, and generally test that the application works as expected.

### Author Checklist
1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? N/A documentation and non-functional code changes
3. Command line build passes? Y
4. Changes have been smoke-tested? Y
5. Testing instructions included? Y

### CODE REVIEW

**Comment by JoelMcKinnon** [comment]
There's a dup in the rules list of https://eslint.org/docs/rules/no-multiple-empty-lines

File: .eslintrc.js [hunk 1]
```diff
@@ -66,6 +67,108 @@ module.exports = {
...
+        //https://eslint.org/docs/rules/no-whitespace-before-property
+        "no-whitespace-before-property": "error",
+        // https://eslint.org/docs/rules/object-curly-newline
+        "object-curly-newline": ["error", {"consistent": true, "multiline": true}],
+        // https://eslint.org/docs/rules/object-property-newline
+        "object-property-newline": "error",
+        // https://eslint.org/docs/rules/brace-style
+        "brace-style": "error",
+        // https://eslint.org/docs/rules/no-multiple-empty-lines
+        "no-multiple-empty-lines": ["error", {"max": 1}],
```
Comment (JoelMcKinnon) [inline .eslintrc.js:152]: This rule is specified above without the `{"max": 1}` qualifier. Is that intended?

File: platform/commonUI/edit/test/actions/SaveAsActionSpec.js [hunk 2]
```diff
@@ -50,7 +50,7 @@ define(
                         catch: function (callback) {
                             return mockPromise(callback(value));
                         }
-                    }   ;
+                    } ;
```
Comment (JoelMcKinnon) [inline platform/commonUI/edit/test/actions/SaveAsActionSpec.js:53]: Why is there still a single space before the semicolon? Shouldn't this be a violation?

**Comment by akhenry** [comment]
Actually, hold off on this, I've just thought of a strategy for making this PR easier to review.

**Comment by JoelMcKinnon** [comment]
> Actually, hold off on this, I've just thought of a strategy for making this PR easier to review.

One thing that occurred to me is if you did one PR for just the multi-line and spacing fixes that would eliminate several hundred files from PRs for subsequent fixes. I think the spacing issues could be assumed to not break functionality.

File: .eslintrc.js (hunk 1 above)
Comment (akhenry) [reply .eslintrc.js:151]: Good catch, thanks! I've fixed it.

**Comment by akhenry** [comment]
Have broken this down into a couple of commits:
1. https://github.com/nasa/openmct/pull/3058/commits/2d0ebc6bceac0b19b5b7fa2150899f704cd70746 deals with code changes for compliance with the new `no-implicit-coercion` and `no-unneeded-ternary` rules.
2. https://github.com/nasa/openmct/pull/3058/commits/6e27b419621104bc3bca3f79d7410f044eade3ac Enforces a semicolon at the end of every statement
3. https://github.com/nasa/openmct/pull/3058/commits/809ac9b8d244bfc8761a411106981c56a7b4ff4a Adds a number of rules that affect spacing and formatting and can be safely assumed not to cause any functional changes.
4. https://github.com/nasa/openmct/pull/3058/commits/e84440db6706ecbf4dfb35584fd28e000f82da0c Enables a rule enforcing spacing around semi-colons.

Again, none of these changes should cause any functional changes, but there is always the risk of a bug in ESLint causing unintended consequences.

**Comment by akhenry** [comment]
> LGTM. I'm not a fan of semi-colons... but I can live with it. Also, I've seen at least one edge case instance in our codebase where not having a semi-colon would have an unexpected side effect.

I went back and forth on this rule a bit, and read some justifications for and against. On the one hand, it's a lot of visual noise, which I dislike. On the other hand, if you don't add semicolons [the JavaScript interpreter will be left to guess where they should be](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Automatic_semicolon_insertion), which can lead to errors.

**Comment by shefalijoshi** [comment]
@akhenry Please resolve conflicts.

**Comment by akhenry** [comment]
@shefalijoshi This is good to go now. We're out of code freeze for VISTA 4.5 so it should be good to merge.

File: src/plugins/timeConductor/ConductorAxis.vue [hunk 3]
```diff
@@ -207,9 +207,9 @@ export default {
             this.$emit('panAxis', panBounds);
         },
         endPan() {
-            const panBounds = this.dragStartX && this.dragX && this.dragStartX !== this.dragX
-                ? this.getPanBounds()
-                : undefined;
+            const panBounds = this.dragStartX && this.dragX && this.dragStartX !== this.dragX ?
```
Comment (nikhilmandlik) [inline src/plugins/timeConductor/ConductorAxis.vue:210]: this formatting feel weird to me, but are we going forward with this? (might take some time for me to get used to it :P )

File: platform/import-export/src/actions/ImportAsJSONAction.js [hunk 4]
```diff
@@ -156,7 +158,7 @@ define(['zepto', 'objectUtils'], function ($, objectUtils) {
                 Object.prototype.hasOwnProperty.call(value, 'namespace') &&
                 value.key === oldId.key &&
                 value.namespace === oldId.namespace) {
-                return newId
+                return newId;
```
[the remark this answers is not in the document]
Comment (akhenry) [reply platform/import-export/src/actions/ImportAsJSONAction.js:161]: It's a single line code block, I don't think that a line break would make sense here.

File: src/plugins/timeConductor/ConductorAxis.vue (hunk 3 above)
Comment (akhenry) [reply src/plugins/timeConductor/ConductorAxis.vue:210]: I am personally more accustomed to `after`, and it is the default configuration for eslint which is why it was set to `after`. After reading a few examples online, I think `before` is actually more readable though (and AirBnB use `before`).

I really want to get this merged ASAP, so in this case I am going to switch to `before`, and let's see how we go, with an exception for `=`. I don't think `=` should ever start a line?

Comment (JoelMcKinnon) [reply src/plugins/timeConductor/ConductorAxis.vue:210]: I like `after` but maybe that's just me. I can see the argument for `before`. It doesn't seem logically consistent, though, to make an exception for `=`.

Comment (davetsay) [reply src/plugins/timeConductor/ConductorAxis.vue:210]: I think this will be obsoleted anyways @nikhilmandlik . #3246

**Comment by akhenry** [comment]
@nikhilmandlik Updated, can you review and merge?

**Pull request merged**